syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.graphsearch";
option java_outer_classname = "GraphSearchProto";
option objc_class_prefix = "GS";

package graphsearch;

service GraphSearchService {
  rpc GraphSearch (GraphSearchRequest) returns (GraphSearchReply) {}
  rpc TrustedValidation (TrustedValidationRequest) returns (TrustedValidationReply) {}
  rpc Status (StatusRequest) returns (StatusReply) {}
}

message GraphSearchRequest {
    string txid = 1;
    repeated string exclude_txids = 2;
}

message GraphSearchReply {
    repeated bytes txdata = 1;
}

message TrustedValidationRequest {
    string txid = 1;
}

message TrustedValidationReply {
    bool valid = 1;
}

message StatusRequest {
}

message StatusReply {
    uint32 block_height = 1;
    string best_block_hash = 2;

    uint64 last_incoming_zmq_tx_unix  = 3;
    uint64 last_outgoing_zmq_tx_unix  = 4;
    string last_incoming_zmq_tx       = 5;
    string last_outgoing_zmq_tx       = 6;

    uint64 last_incoming_zmq_blk_unix = 7;
    uint64 last_outgoing_zmq_blk_unix = 8;
    uint64 last_incoming_zmq_blk_size = 9;
    uint64 last_outgoing_zmq_blk_size = 10;
}
